webpack/assets/javascripts/react_app/common/i18nProviderWrapperFactory.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { IntlProvider } from 'react-intl';
+import i18n from './i18n';
+import helpers from './helpers';
+
+const i18nProviderWrapperFactory = initialNow =>
+  (WrappedComponent) => {
+    const wrappedName = helpers.getDisplayName(WrappedComponent);
+
+    class I18nProviderWrapper extends React.Component {
+      constructor(props) {
+        super(props);
+        this.state = { i18nLoaded: i18n.loaded };
+
+        if (!i18n.loaded) {
+          i18n.ready.then(() => {
+            this.setState({ i18nLoaded: true });
+          });
+        }
+      }
+
+      render() {
+        if (!this.state.i18nLoaded) {
+          return <span />;
+        }
+        return (
+          <IntlProvider locale={i18n.locale} initialNow={initialNow}>
```

Comment:
I think you can render messages on the `IntlProvider` as well...

https://egghead.io/lessons/react-convert-a-hard-coded-string-using-react-intl-formattedmessage

Reply:
Yeah, I know about it. But we don't use react-intl for messages. It's one of the options for future, but we haven't decided what will be our approach yet. We can easily add it later.